app/controllers/api/ladder_matches_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Api::LadderMatchesController < ApplicationController
+  def index
+    matches = LadderMatch.all
+    render json: {
+      matches: JSON.parse(matches.to_json())
+    }
+  end
+  def create
+    p = ActionController::Parameters.new params[:ladder_match]
+    permitted = p.permit(:player1,:player2, :date_of_match, :winner)
+    match = LadderMatch.new permitted
+    @player1 = LadderUser.find_by(username: p[:player1])
+    match.password = p[:password]
+	#authentication and validations on serverside
+    unless match.valid? && @player1.try(:authenticate, match.password)
```

Comment:
I had to handle validation of password and saving of model data separately. This way, the authentication would not be saved as a model attribute.

The logic repeated itself for error messages.